grand_challenge_forge/partials/example-evaluation-method/example-evaluation-method{{cookiecutter._}}/helpers.py

```diff
@@ -100,15 +100,21 @@ def _start_pool_worker(fn, predictions, max_workers, results, errors):
 
 def _pool_worker(*, fn, predictions, max_workers, results, errors):
     terminating_child_processes = False
-
+    executor_shutting_down = False
```

Comment:
After a LOT of testing locally on a flaky run-through, I had determined that the `shutdown()` call would sometimes hang. After pushing it once... the test kept failing, but today they no longer do.

I've spent too much time debugging this and will just accept that it is flaky at the moment.

Reply:
Err, that's not good. If it is flaky then we are definitely going to hit it in production.

Reply:
Sharing your worries I've dived a bit deeper into what the possible cause could be. It seems that the signal handler makes things unstable. It's not quite an async context but:

> A Python signal handler does not get executed inside the low-level (C) signal handler. Instead, the low-level signal handler sets a flag which tells the [virtual machine](https://docs.python.org/3/glossary.html#term-virtual-machine) to execute the corresponding Python signal handler at a later point(for example at the next [bytecode](https://docs.python.org/3/glossary.html#term-bytecode) instruction).

Not sure how this interacts with the GIL, but I can imagine the normal flow gets interrupted. Since the actual handler interacted with the executors `_shutdown_lock` it messes with locks. If a signal handler got executed while a shutdown_lock was active, it would lock up the main thread: pending a timeout.

The executor seems to raise an exception when one of pool workers is killed. Hence, I don't think we even need the signal handler in the first place. Luckily, that also entails we can simplify the worker exception handling.